lib/engine/game/g_1822_pnw/step/merge.rb

```diff
@@ -0,0 +1,214 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+require_relative '../../../token'
+require_relative '../../../step/token_merger'
+require_relative '../../../step/programmer_merger_pass'
+
+module Engine
+  module Game
+    module G1822PNW
+      module Step
+        class Merge < Engine::Step::Base
+          include Engine::Step::TokenMerger
+          include Engine::Step::ProgrammerMergerPass
+
+          def actions(entity)
+            return [] if !entity.corporation? || entity != current_entity || @merge_state == :done
+            return ['choose'] if @merge_state != :none
+            return ['merge'] if mergeable(entity).any?
+
+            []
+          end
+
+          def setup
+            @merge_state = :none
+          end
+
+          def merge_name(_entity = nil)
+            'Merge'
+          end
+
+          def description
+            'Merge Corporation'
+          end
+
+          def mergeable(corporation)
+            @game.unassociated_minors.select do |m|
+              @game.graph.connected_nodes(corporation)[m.tokens.first.city] && !valid_par_prices(corporation, m).empty?
+            end
+          end
+
+          def choice_name
+            case @merge_state
+            when :selecting_par
+              'Choose par value for new company'
+            when :selecting_shares
+              'Choose number of shares'
+            when :selecting_token
+              'What to do with the token'
+            end
+          end
+
+          def min_exchange_shares(par, minors_value, minors_cash)
+            ((minors_value - minors_cash - 1) / par).floor + 1
```

Comment:
I believe If you don't have a float, integer math will be used.

Reply:
It will in ruby, yes - but since it is just taking the floor at the end, it doesn't actually matter.